src/ansys/units/quantity.py

```diff
@@ -296,10 +297,16 @@ def _relative_unit_check(self, __value, op: operator = operator.add) -> Quantity
     def __float__(self):
         base_dims = BaseDimensions
         dims = Dimensions
+        if os.getenv("PYANSYS_UNITS_ANGLE_AS_DIMENSION"):
+            angle_dims = {base_dims.ANGLE: 1.0}
+            solid_angle_dims = {base_dims.SOLID_ANGLE: 1.0}
+        else:
+            angle_dims = {}
+            solid_angle_dims = {}
         if self.dimensions in [
             dims(),
-            dims(dimensions={base_dims.ANGLE: 1.0}),
-            dims(dimensions={base_dims.SOLID_ANGLE: 1.0}),
+            dims(dimensions=angle_dims),
+            dims(dimensions=solid_angle_dims),
```

Comment:
1. The code looks inaccurate. E.g., what if the "dimensions" are `angle ** 2`?
2. This code is out of place here: we should determine in an object-oriented manner whether the quantity is _physically dimensionless_. The implementation details don't go in the `__float__` conversion method. We can ask the relevant question of the dimension object.
3. Do not check `PYANSYS_UNITS_ANGLE_AS_DIMENSION` more than once in the whole library. It should do its job of configuring the behaviour once and once only.
4. Is an environment variable a suitable configuration tool? Why not use the existing file-based configuration so that the library can be configured to be used out of the box?

Reply:
1. Such units are allowed to exist (as they should be) and therefore we should check for them.
```Python
>>> import ansys.units as au
>>> au.Quantity(1.0, "radian") ** 2
Quantity (1.0, "radian^2")
```
2. In the latest version, it's a hard-coded check for angles (no longer configurable). The original point was about enapsulation. We need to encapsulate the dimensions functionality in a class/module that's concerned with dimensions, not spread the functionality all around.
3. What is the relevance of PyFluent environment variables here? I'm saying that we should not couple many pieces of code to an environment variable when instead we could couple a single variable or function to that environment variable and have everything else depend on that single variable or function. 
4. We are already using (a) config file(s) to control the behaviour of the library.

Reply:
@seanpearsonuk 

1. Right.
2. The existing code should be refactored first with latest suggestion given in point 1 above, as this is a bug that already exists (hard coded {"ANGLE": 1.0} and {"SOLID_ANGLE": 1.0} which do not consider angle ** 2 case.
3. What is the relevance of PyFluent environment variables here? - It was not clear why we can't use env var at more than 1 place which we are already doing in PyFluent repo. Now you have explained the reason so it's clear now.
4. We have recent requirement to [register a unit without need of updating config file](https://github.com/ansys/pyansys-units/pull/282). Therefore I'm avoiding file based configuration approach.

Reply:
We are not handling at this level.